contrib/cirrus/pr-removes-fixed-skips.t

```diff
@@ -0,0 +1,134 @@
+            { name => $1, issues => [], files => {}, expect => [] };
+    }
+    elsif ($line =~ /^\[([\d\s,]+)\]$/) {
+        $full_tests[-1]{issues} = [ split /,\s+/, $1 ];
+    }
+
+    #                  1     1   23   3 4   4 5  52
+    elsif ($line =~ m!^(\!|\+)\s+((\S+):(\d+):(.*))$!) {
+        push @{$full_tests[-1]{expect}}, $2 if $1 eq '+';
+
+        $full_tests[-1]{files}{$3}[$4] = $5;
+    }
+}
+
+plan tests => 1 + 1 + @full_tests;
+
+require_ok "$FindBin::Bin/pr-removes-fixed-skips";
+
+#
+# Parser tests. Just run as one test.
+#
+$ENV{CIRRUS_CHANGE_MESSAGE} = '';
+my @parser_expect;
+for my $line (split "\n", $parser_tests) {
+    if ($line =~ s/\s+\|\s+([\d\s]+)$//) {
+        push @parser_expect, split ' ', $1;
+    }
+    $ENV{CIRRUS_CHANGE_MESSAGE} .= $line . "\n";
+}
+
+my @parsed = Podman::CI::PrRemovesFixedSkips::fixed_issues();
+eq_or_diff \@parsed, \@parser_expect, "parser parses issue IDs";
+
+###############################################################################
+
+#
+# Full tests. Create dummy source-code trees and verify that our check runs.
+#
+my $tmpdir = tempdir(basename($0) . ".XXXXXXXX", TMPDIR => 1, CLEANUP => 1);
+chdir $tmpdir
+    or die "Cannot cd $tmpdir: $!";
+mkdir $_        for qw(cmd libpod pkg test);
+for my $t (@full_tests) {
+    for my $f (sort keys %{$t->{files}}) {
+        my $lineno = 0;
+        make_path(dirname($f));
+        open my $fh, '>', $f or die;
+
+        my @lines = @{$t->{files}{$f}};
+        for my $i (1 .. @lines + 10) {
+            my $line = $lines[$i] || "[line $i intentionally left blank]";
+            print { $fh } $line, "\n";
+        }
+        close $fh
+            or die;
+    }
+
+    # FIXME: run test
+    my @actual = Podman::CI::PrRemovesFixedSkips::unremoved_skips(@{$t->{issues}});
+    eq_or_diff \@actual, $t->{expect}, $t->{name};
+
+    # clean up
+    unlink $_   for sort keys %{$t->{files}};
+}
+
+chdir '/';
+
+__END__
+
+== basic test
+[12345]
+! test/foo/bar/foo.bar:10:   skip "#12345: not a .go file"
++ test/foo/bar/foo.go:17:    skip "#12345: this one should be found"
++ test/zzz/foo.bats:10:   # FIXME: #12345: we detect FIXMEs also
+
+== no substring matches
+[123]
+! test/system/123-foo.bats:12:    skip "#1234: should not match 123"
+! test/system/123-foo.bats:13:    skip "#0123: should not match 123"
+
+== multiple matches
+[456, 789]
++ cmd/podman/foo_test.go:10:    Skip("#456 - blah blah")
+! cmd/podman/foo_test.go:15:    Skip("#567 - not a match")
++ cmd/podman/foo_test.go:19:    Skip("#789 - match 2nd issue")
++ cmd/podman/zzz_test.go:12:    Skip("#789 - in another file")
+
+== no match on bkp files
+[10101]
+! pkg/podman/foo_test.go~:10:    Skip("#10101: no match in ~ file")
+! pkg/podman/foo_test.go.bkp:10:    Skip("#10101: no match in .bkp file")
+
+== no match if Skip is commented out
+[123]
+! test/e2e/foo_test.go:10:   // Skip("#123: commented out")
+! test/system/012-foo.bats:20:      # skip "#123: commented out"
```

Comment:
This does not seem to work correctly for all cases. This test will fail:
```
! test/e2e/foo_test.go:10:   // Skip("FIXME: #123 commented out")
! test/system/012-foo.bats:20:      # skip "FIXME: #123 commented out"
```

But do we actually want to exclude commented skips? Skips should be removed and not commented out, especially when the issues is supposedly fixed.

Reply:
Again, I was going for quick-easy. To understand the problem  with comments we have to go back to the `grep` expression in line 129 above. I found it safest to check for:
* `skip` with only leading whitespace
* `FIXME` anywhere on the line

I can't find any instances right now like '// make sure we don't skip bytes (#11496)` in the code itself, only in release notes, but I could imagine someone adding that and getting very upset about being blocked for it. Unlikely, I know.

Again, this is not meant to be perfect. I do manual scans of the repo every once in a while, with a much looser expression, because I don't mind manually double-checking the questionable cases. The purpose of this is simply to make my job a little easier, so I can spend my days sipping margaritas on the beach.